doc: release-notes-latest: added NCS specific extensions to Zephyr (#2667)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-latest.rst

The following list contains |NCS| specific additions:

> * The Bluetooth controllers default uses split architecture

Reviewer: What does this mean? What is "split architecture" - split between Host and Controller?
And why "controllers" - doesn't Zephyr have only one Controller?

Maybe something like:
```suggestion
* Configured the Zephyr Bluetooth stack to build only the Host or only the Controller instead of doing a combined build (see :ref:`zephyr:bluetooth-build-types`).
  The reason for this default configuration is to add support for the :ref:`ble_controller`.
```
The links point to https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/bluetooth/bluetooth-arch.html#build-types and https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrfxlib/ble_controller/README.html#nrf-bluetooth-le-controller , respectively.

Author: That should actually be described in the docs, although that section seems a bit empty:
https://docs.zephyrproject.org/latest/guides/bluetooth/bluetooth-arch.html#split

Author: and here:
http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/kconfig/CONFIG_BT_LL_SW_SPLIT.html#cmdoption-arg-config-bt-ll-sw-split

Reviewer 2: What is the purpose here? To describe ways that NCS is different from Zephyr? In that case I think until the SoftDevice controller is the default this paragraph can be removed.

@rugeGerritsen Any opinions here?

Reviewer 2: Note that Legacy (as opposed "split") has been removed, so Zephyr only has one choice symbol here.

@cvinayak @carlescufi https://docs.zephyrproject.org/latest/guides/bluetooth/bluetooth-arch.html#split
This could perhaps be remove as well? Only one option now.

Reviewer: > What is the purpose here? To describe ways that NCS is different from Zephyr?

The idea is to point out persistent differences between NCS's Zephyr fork and upstream Zephyr.

Reviewer 2: Right, so remove this sentence then. Since this is not different for upstream.

Reviewer 3: Yes, please remove this @tejlmand

Author: Then why do we still have this path ?
https://github.com/nrfconnect/sdk-zephyr/commit/e85c858c21752b50b5b9e6f26cf78300c168a882

Reviewer 2: @tejlmand Because we add another option to that menu, and since order matters we need the default statement there to keep the same default.

Author: got it.
Thanks for explaining :)

> * NCS Partition Manager (PM) for flash partitioning

Reviewer: ```suggestion
* Added support for the |NCS|'s :ref:`partition_manager`, which can be used for flash partitioning.
```

> * NCS network socket and address extensions:
>   AF_LTE, NPROTO_AT, NPROTO_PDN, NPROTO_DFU, SOCK_MGMT, SO_RCVTIMEO, SO_BINDTODEVICE, SOL_PDN, SOL_DFU, SO_PDN_CONTEXT_ID, SO_PDN_STATE, SOL_DFU, SO_DFU_ERROR, TLS_SESSION_CACHE,
>   SO_SNDTIMEO, MSG_TRUNC
>   SO_SILENCE_ALL, SO_IP_ECHO_REPLY, SO_IPV6_ECHO_REPLY

Reviewer: ```suggestion
* Added the following network socket and address extensions to the :ref:`zephyr:bsd_sockets_interface` interface to support the functionality provided by the :ref:`nrfxlib:bsdlib`:
  AF_LTE, NPROTO_AT, NPROTO_PDN, NPROTO_DFU, SOCK_MGMT, SO_RCVTIMEO, SO_BINDTODEVICE, SOL_PDN, SOL_DFU, SO_PDN_CONTEXT_ID, SO_PDN_STATE, SOL_DFU, SO_DFU_ERROR, TLS_SESSION_CACHE, SO_SNDTIMEO, MSG_TRUNC, SO_SILENCE_ALL, SO_IP_ECHO_REPLY, SO_IPV6_ECHO_REPLY
```
Is this what they are needed for?

Author: Those are the exact extensions added.
Listing them here makes it much easier to search for them, if they are not already know to the reader.
(some are nordic specific)

Reviewer 4: >  Listing them here makes it much easier to search for them, if they are not already know to the reader.

But how can we avoid this going stale?

> * MQTT: Support enabling of TLS caching

Reviewer: ```suggestion
* Added support for enabling TLS caching when using the :ref:`zephyr:mqtt_socket_interface` library.
```
How do you enable it?

Author: added  link.

> * CC310: Cryptocell access from non-secure code.

Reviewer: ?
Where was that added? What is the actual change - support for CryptoCell? Or support for access from non-secure code?

Author: Support for access from non-secure code:
https://github.com/nrfconnect/sdk-zephyr/commit/32c5d847e8d0196e9b9a32413c67cc98089b1ef2

Reviewer: ```suggestion
* Updated the nrf9160ns DTS to support accessing the CryptoCell CC310 hardware from non-secure code.
```
Does that make any sense?

The following list contains |NCS| specific additions:

* Added support for the |NCS|'s :ref:`partition_manager`, which can be used for flash partitioning.
* Added the following network socket and address extensions to the :ref:`zephyr:bsd_sockets_interface` interface to support the functionality provided by the :ref:`nrfxlib:bsdlib`:
  AF_LTE, NPROTO_AT, NPROTO_PDN, NPROTO_DFU, SOCK_MGMT, SO_RCVTIMEO, SO_BINDTODEVICE, SOL_PDN, SOL_DFU, SO_PDN_CONTEXT_ID, SO_PDN_STATE, SOL_DFU, SO_DFU_ERROR, TLS_SESSION_CACHE, SO_SNDTIMEO, MSG_TRUNC, SO_SILENCE_ALL, SO_IP_ECHO_REPLY, SO_IPV6_ECHO_REPLY

> * Added support for enabling TLS caching when using the :ref:`zephyr:mqtt_socket_interface` library. See also `Enabling and testing TLS in mqtt_simple`_

Reviewer: Please put the second sentence on a new line.

That link explains how to enable TLS, but is this what we mean with "TLS caching"? Then we should maybe word it differently, because it sounds like this is something different than what is described in the blog post. Or is there a new Kconfig option (or some other configuration) for TLS caching?

Author: > That link explains how to enable TLS, but is this what we mean with "TLS caching"?

Actually not. I just browsed the code.
I can only see that the option is now possible, but I don't find any place where we actual use this feature.
Maybe we should leave it out until we actual have a sample that shows how this can be used.
@jtguggedal any input on this ?

Will remove link again, as it only describes TLS itself and not the caching option.

Reviewer 5: TLS caching option has been added to the downstream version of the MQTT library (no support in native network stack, so can't be added upstream yet), so its use is shown there:
https://github.com/nrfconnect/sdk-zephyr/blob/26216d1f7e09e9c43dbcfaf0a8565bb0f69d0130/subsys/net/lib/mqtt/mqtt_transport_socket_tls.c#L81

I have a new library in the making that uses the socket option, and it will hopefully feature in NCS v1.4.0.
I can also add TLS caching to the MQTT sample in time for NCS 1.4.0 to show how it can be used.

Author: > TLS caching option has been added to the downstream version of the MQTT library (no support in native network stack, so can't be added upstream yet), so its use is shown there:

Well, that is also the code I found, but that just checks if the config enables it, and then sets the sock option accordingly.

What I referred to was somewhere that a sample was doing, for examlple
```
tls_config->session_cache = TLS_SESSION_CACHE_ENABLED;
```
similar to:
https://github.com/nrfconnect/sdk-nrf/blob/master/samples/nrf9160/aws_fota/src/main.c#L431-L438

and I don't find such example anywhere.
Therefore the question, if we should leave it out.

> I have a new library in the making that uses the socket option, and it will hopefully feature in NCS v1.4.0.
> I can also add TLS caching to the MQTT sample in time for NCS 1.4.0 to show how it can be used.

When that is available, then we should probably update release notes, and thus leave it out for now?

Reviewer 5: If it's practice to leave things out if it's not shown how to use it in a sample, I'm fine with that.
We will highlight TLS caching more in v1.4.0 and make sure it goes into the release notes then.

Reviewer: There's no common practice since this is the first time we're adding these release notes. ;)
But that option is actually documented, so how about:
```suggestion
* Added support for enabling TLS caching when using the :ref:`zephyr:mqtt_socket_interface` library.
  See :c:macro:`TLS_SESSION_CACHE`.
```
This will then link to https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/reference/networking/sockets.html?highlight=tls_session_cache#c.TLS_SESSION_CACHE

Author: Done

The following list contains |NCS| specific additions:

* Added support for the |NCS|'s :ref:`partition_manager`, which can be used for flash partitioning.
* Added the following network socket and address extensions to the :ref:`zephyr:bsd_sockets_interface` interface to support the functionality provided by the :ref:`nrfxlib:bsdlib`:
  AF_LTE, NPROTO_AT, NPROTO_PDN, NPROTO_DFU, SOCK_MGMT, SO_RCVTIMEO, SO_BINDTODEVICE, SOL_PDN, SOL_DFU, SO_PDN_CONTEXT_ID, SO_PDN_STATE, SOL_DFU, SO_DFU_ERROR, TLS_SESSION_CACHE, SO_SNDTIMEO, MSG_TRUNC, SO_SILENCE_ALL, SO_IP_ECHO_REPLY, SO_IPV6_ECHO_REPLY

> * Added support for enabling TLS caching when using the :ref:`zephyr:mqtt_socket_interface` library.

Reviewer: ```suggestion
* Added support for enabling TLS caching when using the :ref:`zephyr:mqtt_socket_interface` library.
  See :c:macro:`TLS_SESSION_CACHE`.
```

The following list contains |NCS| specific additions:

* Added support for the |NCS|'s :ref:`partition_manager`, which can be used for flash partitioning.
* Added the following network socket and address extensions to the :ref:`zephyr:bsd_sockets_interface` interface to support the functionality provided by the :ref:`nrfxlib:bsdlib`:

>   AF_LTE, NPROTO_AT, NPROTO_PDN, NPROTO_DFU, SOCK_MGMT, SO_RCVTIMEO, SO_BINDTODEVICE, SOL_PDN, SOL_DFU, SO_PDN_CONTEXT_ID, SO_PDN_STATE, SOL_DFU, SO_DFU_ERROR, TLS_SESSION_CACHE, SO_SNDTIMEO, MSG_TRUNC, SO_SILENCE_ALL, SO_IP_ECHO_REPLY, SO_IPV6_ECHO_REPLY

Reviewer 4: I think we need some documentation added to our internal synchronization procedures to keep this list updated at each upmerge too.

Author: agree.
